include/flash.h

```diff
@@ -107,13 +121,9 @@ static inline int flash_erase(struct device *dev, off_t offset, size_t size)
 /**
  *  @brief  Enable or disable write protection for a flash memory
  *
- *  This API is required to be called before the invocation of write or erase
- *  API. Please note that on some flash components, the write protection is
- *  automatically turned on again by the device after the completion of each
- *  write or erase calls. Therefore, on those flash parts, write protection needs
- *  to be disabled before each invocation of the write or erase API. Please refer
- *  to the sub-driver API or the data sheet of the flash component to get details
- *  on the write protection behavior.
```

Comment:
I raised this question vi zephyr-devel mailing [Zephyr-devel] Flash drivers compatibility attitude:
>It is unwanted freedom because it is the driver API and hence it should 
>provide abstraction over HW e.g write operation could perform a write 
>of a long buffer despite that HW write is word-by-word operation so HW 
>protect must be served by a such driver automatically and protection 
>offered by this API should be the abstraction over HW.

as David Brown answered
>I think the solution here should be to fix the wording of the API, since it makes the driver almost useless for normal operation.

At last nRF5x driver work in this manner.
I'm not sure how the modification process should looks like for such a change.